docs/tools/mdcode.py

```diff
@@ -27,13 +27,18 @@ def _parse_lines(self):
             m = _CODE_MARK.match(l)
             if m is not None:
                 if not in_code:
-                    lang = m.groups()[1].lower()
-                    indent = len(m.groups()[0])
-                yield (l, True, lang, indent)
-                if in_code:
+                    if _LANGS.has_key(m.groups()[1].lower()):
```

Comment:
do we need to limit to the langs defined in _LANGS? i don't suggest to because it may confuse users.